Devbook: upgrade crates and openapi script (#2042)
input-output-hk/mithril: docs/devbook/upgrade-crates-and-openapi-versions/README.md

# Upgrade the crates and openapi versions before merging a pull request

## Introduction

> This devbook provides a script that allow to automatically upgrade the crates and openapi versions in the project.

Reviewer: ```suggestion
This devbook provides a script that allows to automatically upgrade the crates and Open API versions in the project.
```

> Only crates and openapi with changes on the branch compared to `origin/main` are upgraded.

Reviewer: ```suggestion
Only the crates and Open API specifications with changes on the branch compared to `origin/main` are upgraded.
```

## Prerequisites

> It requires having `cargo-get` installed, which can be done with the following command:

Reviewer: ```suggestion
It requires to have `cargo-get` installed, which can be done with the following command:
```

```
cargo install cargo-get
```

## Usage

> [!NOTE]
> All commands are executed from the root of the repository.

### Dry-run

Just run the script without argument, by default no changes are made to the project.

```shell
. ./docs/devbook/upgrade-crates-and-openapi-versions/upgrade_crates_and_openapi_versions.sh
```

### Run

Run the script with the `--run` argument to bump the crates and openapi versions.

> The script will finish with a preformatted commit message that can be used to create a commit.

Reviewer: ```suggestion
The script will output a preformatted commit message that can be used to create a commit when it completes.
```
